### PR TITLE
Lost validation code fix

### PR DESCRIPTION
**Pull request by KiranPrabhakar135**
Fixed the lost validation code page load issue.
Modified the logic to remove the duplicate redirect to the same route. This removes the error in the console for redirect to same url.
Altered the logic for getting the user back to sign page if the user has not chosen the remember me chk box.
Introduced new friendly route for forgot password.
Fixed the broken back to sign button functionality in Forgot password, register and lost validation code components.
made changes to code to avoid redundant verify email check  when registering the user.

### CODE REVIEW

**Review by aashish-jain — APPROVED** [review]
Looks good. Please merge after Linus / Yifu have a look into it.

**Review by LinusCastelino — APPROVED** [review]
Good job with clearing the quirky flags. Comments made it easy to understand.
Remember accounts feature needs rework - but that will be a different task.
Approved.

Suggestion - the variable 'validateEmailPressed' is misleading. Please rename to 'lostValidationCodePressed'

File: src/main/react-front-end/src/views/Login/CreateAccountComponent.js [hunk 1]
```diff
@@ -197,7 +189,7 @@ export default class CreateAccountComponent extends Component {
 
           <ValidatorForm
             ref="email"
-            onSubmit={this.onEmailNextClicked}>
```
Comment (LinusCastelino) [inline src/main/react-front-end/src/views/Login/CreateAccountComponent.js:200]: We need to check if the email is already present in the database before registering incase the user tries to create an account with an already existing email address. Please revert this change.

**Review by LinusCastelino — CHANGES_REQUESTED** [review]

**Comment by KiranPrabhakar135** [comment]
Incorporated review comments

**Review by LinusCastelino — APPROVED** [review]
Looks good to merge

**Pull request merged**